src/System.Management.Automation/engine/SessionStateLocationAPIs.cs

```diff
@@ -231,10 +232,36 @@ internal PathInfo SetLocation(string path, CmdletProviderContext context)
             ProviderInfo provider = null;
             string providerId = null;
 
+            // Replace path with last working directory when '-' was passed.
+            if (_workingLocationHistoryStack == null)
+            {
+                const int locationHistoryLimit = 1000;
```

Comment:
I think 1000 is at least 1, maybe 2 orders of magnitude too large.
Many people will never use this feature, and those that do will struggle to unwind the stack > [7](https://en.wikipedia.org/wiki/The_Magical_Number_Seven,_Plus_or_Minus_Two) anyway, instead using some other tool like [ZLocation](https://github.com/vors/ZLocation) or something like [this](https://github.com/lzybkr/PSReadLine/blob/d3e05af42dc550a31a44ac7f8b464e4b47094788/PSReadLine/SamplePSReadlineProfile.ps1#L504).

Reply:
Ok. Shall we go for 50 then? This number also probably needs to go into the docs afterwards.

Reply:
I was thinking 10, maybe 15.

Reply:
I can and will put in any number you'd like me to but I don't understand why we need to do penny pinching for a bunch of strings that are usually just around 100 char long. In terms of usage I can imagine some people including me wanting to go back further (I typically step out of directories one at a time in PS because I just repeatedly press 'up' and 'enter' after the first `cd ..`)

Reply:
I go over 10 frequently enough with the PSCX impl of this.  I was down with the `50` suggestion but at least make it 20.  :-)

Reply:
Maybe Jeffrey should post a survey about this on Twitter (JK). I think it is an overkill but if we really need a low default then we could offer the user to override the default value by setting an environment variable or automatic variable?

Reply:
An configuration setting is probably overkill.

You're storing a PathInfo, not a string. It's not immediately obvious to me that there are no NoteProperties also being stored - another small concern for the storage costs.

At any rate, I see assumptions far too often that something costs nothing and we find out far later that the assumption was wrong. Or it might have been right at the time, but something else crept in and made the other thing more expensive.

I try to be conservative because I tend to be the person finding those sorts of issues - the folks that make these assumptions tend to never see the consequences.

I'm fine with 20.